lblod/reglement-publish-service
# Add proper authorization checks

### Overview
This PR ensures that the endpoints of this service may only be used by logged-in and properly authorized users.
Specifically this PR includes:
- A middleware which checks if the user is logged in and has the proper access rights. Returns a 401 error otherwise
- Replacement of `mu-auth-sudo` queries by queries employing the standard access rights.

##### connected issues and PRs:
To be used with https://github.com/lblod/app-reglementaire-bijlage/pull/89


### Setup
Include this service in your stack:
```yaml
publisher:
  image: !reset null
  build: https://github.com/lblod/reglement-publish-service.git#improve-auth
```

### How to test/reproduce
- Start-up the stack
- Open the frontend and ensure you can publish a template
- Check the following:
   * Create tasks should now be stored in the `organization` graph, no longer in the `application` graph
   * The create `Template` and `TemplateVersion` resources should be stored in the public graph
- Ensure that you may not use the endpoints when not logged in. Without this PR, it was possible to publish a template without being logged in.

### Challenges/uncertainties
If we can avoid it, I think we should prevent using `mu-auth-sudo` queries to access/update the database in services.

### Checks PR readiness
